### DISCUSSION TITLE
Why does ComfyUI downgrade PyTorch when it updates?

### DISCUSSION THREAD

**Discussion by usul12** [comfyanonymous/ComfyUI#4795] [General]
Everytime I do an update I have to go afterwards and reinstall PyTorch 2.4.0 to use xFormers because it downgrades it to 2.1.2. 

Am I breaking something else by doing that?

**Comment by ltdrdata** [comment]
This phenomenon occurs due to some custom nodes.
ComfyUI-Manager has taken measures to prevent the reinstallation of torch dependencies during the general dependency installation process. However, some custom nodes are set to install dependencies on their own, so this method cannot be applied to them.
It's necessary to check the entire log at the time the downgrade occurred to identify the issue.

**Reply by usul12** [reply]
Thanks for the reply. I'll do some digging through the logs.

**Reply by lichenophile** [reply]
Given the version it's downgrading to, I'm going to assume that you might be using Stability Matrix and this isn't caused by a custom node at all. 

https://github.com/LykosAI/StabilityMatrix/blob/f4e3cbeddd2cca659a1d9e22f78d22bf233bebd5/CHANGELOG.md?plain=1#L68
```
## v2.11.4: 
- Fixed ComfyUI slower generation speed with new torch versions not including flash attention for windows, pinned `torch==2.1.2` for ComfyUI on Windows CUDA
```

I have tested speeds on newer versions and noticed no slower generation, so I have to assume the above is now pointless and needs removing.

Related issue: https://github.com/LykosAI/StabilityMatrix/issues/861
